prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,645 @@
+        if sub1 in N_dict:
+            N_dict[sub1].append(sub2)
+        else:
+            nodes.append(sub1)
+            N_dict[sub1] = [sub2]
+        if sub2 in N_dict:
+            N_dict[sub2].append(sub1)
+        else:
+            nodes.append(sub2)
+            N_dict[sub2] = [sub1]
+    return lines, nodes, N_dict
+
+
+def lineFromCSV(T_csv):
+    """Create dict to store all the raw transmission line csv data
+
+    :param str T_csv: path of the HIFLD transmission csv file
+    :return: (*dict*) -- a dict mapping the transmission ID to its raw parameters.
+    """
+
+    raw_data = pd.read_csv(T_csv)
+    raw_data["ID"] = raw_data["ID"].astype("str")
+    raw_lines = raw_data.set_index("ID").to_dict()
+    return raw_lines
+
+
+def meter2Mile(dist):
+    """Calculate the mile given the distance in meter
+
+    :param float dist: length of the distance between two substations
+    :return: (*float*) -- distance in mile unit.
+    """
+
+    return dist / 1609.34
+
+
+def computeGeoDist(sub1, sub2):
+    """Compute the haversine geo distance of two substations
+
+    :param tuple sub1: (latitude, longitude) of substation1
+    :param tuple sub2: (latitude, longitude) of substation2
+    :return: (*float*) -- distance between two geo coordinates.
+    """
+
+    return haversine(sub1, sub2, Unit.MILES)
+
+
+def GraphOfNet(nodes, N_dict):
+    """Create the graph for the HIFLD transmission network
+
+    :param list nodes: list of substation as the vertex inside the Graph
+    :param dict N_dict: dict of substation's neighbors as the edge inside the Graph
+    :return: (*networkx.Graph*) -- Graph generated by networkx.
+    """
+
+    G = nx.Graph()
+    for node in nodes:
+        G.add_node(node)
+
+    for key in N_dict:
+        for value in N_dict[key]:
+            G.add_edge(key, value)
+
+    return G
+
+
+def GetMaxIsland(G):
+    """Report the largest connected island to understand the topo nature of the graph
+
+    :param networkx.Graph G: Graph generated by :func:`GraphOfNet`
+    :return: (*list*) -- list of nodes inside the largest island.
+    """
+
+    Max_nodeSet = []
+    Max_size = 0
+    # list1 = []
+    num = 0
+    for c in nx.connected_components(G):
+        num = num + 1
+        nodeSet = G.subgraph(c).nodes()
+        size = len(nodeSet)
+        # list1.append(size)
+        if size > Max_size:
+            Max_nodeSet = nodeSet
+            Max_size = size
+    # list1.sort(reverse=True)
+    return Max_nodeSet
+
+
+def InitKV(clean_data):
+    """Calculate the base_KV for each node based on MIN_VOLT and MAX_VOLT;
+    Save the invalid substations as to be calculated later
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`Clean`
+    :return: (*dict*) -- KV_dict, a dict of substation's baseKV value.
+    :return: (*list*) -- to_cal, a list of substations to be calculated using neighbour search.
+    """
+
+    KV_dict = {}
+    to_cal = []
+    Max_Value = 3000
```

Comment:
The data seems to only have substations with max voltages up to 1000 kV. Allowing anything up to 3000 kV seems to be overly lax.

It would be nice to have these as parameters with default values, rather than hardcoded.

Reply:
agree - will use `InitKV(clean_data, Max_value=1100)` as parameter